### PR TITLE
RDM-9577 tests

### PR DESCRIPTION
**Pull request by noronhaa**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-9577

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by danlysiak — APPROVED** [review]
Coverage looks good to me 👍

File: aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/befta/DefinitionStoreTestAutomationAdapter.java [hunk 1]
```diff
@@ -12,6 +13,15 @@
     private TestDataLoaderToDefinitionStore loader = new TestDataLoaderToDefinitionStore(this, DEFAULT_DEFINITIONS_PATH,
             BeftaMain.getConfig().getTestUrl());
 
+    @Override
+    public Object calculateCustomValue(BackEndFunctionalTestScenarioContext scenarioContext, Object key) {
+        if (key.toString().startsWith("no_dynamic_injection_")) {
+            return key.toString().replace("no_dynamic_injection_","");
```
Comment (MSancaktutar) [inline aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/befta/DefinitionStoreTestAutomationAdapter.java:19]: `${}` evaluates to an empty string, already. perhaps no need for this trick?

File: aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/F-094.feature [hunk 2]
```diff
@@ -0,0 +1,38 @@
+@F-094
+Feature: F-094 Retrieve ChallengeQuestions for Notice of Change
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/F-094.feature:2]: `F-094: Get Challenge Questions for NoC` - I suggest.

File: aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/F-094.feature [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    Given an appropriate test context as detailed in the test data source,
+    And a user [MCA System - with an IDAM role of caseworker CAA],
+
+  @S-094.1
+  Scenario: Must successfully return the contents of the ChallengeQuestion definition
+    Given the request [intends to get the contents of a set of questions],
+    And the request [contains a valid CaseType ID],
+    And the request [contains a valid Collection Question ID],
+    And a request is prepared with appropriate values,
+    When it is submitted to call the [GET ChallengeQuestion] operation of [CCD Definition Store Api],
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/F-094.feature:14]: The operation name should be consistent with the name fo the feature, as this feature is itself the operation under test.

File: aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/F-094.feature [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    Then a positive response is received,
+    And the response has all the details as expected.
+
+  @S-094.2
+  Scenario: must return an error response (empty list returned) for a malformed CaseType
+    Given a request is prepared with appropriate values,
+    And the request [intends to get the contents of a set of questions],
+    And the request [contains a malformed CaseType ID for C1],
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/F-094.feature:22]: Placeholder C1 doesn't make direct sense in the context of the scenario flow.

File: aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/F-094.feature [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+Feature: F-094 Retrieve ChallengeQuestions for Notice of Change
+
+  Background:
+    Given an appropriate test context as detailed in the test data source,
+    And a user [MCA System - with an IDAM role of caseworker CAA],
+
+  @S-094.1
+  Scenario: Must successfully return the contents of the ChallengeQuestion definition
+    Given the request [intends to get the contents of a set of questions],
+    And the request [contains a valid CaseType ID],
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/F-094.feature:11]: Better not to use nouns like type names in a programming context.
`And the request [contains a valid Case Type ID],`

File: aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/F-094.feature [hunk 6]
```diff
@@ -0,0 +1,38 @@
+@F-094
+Feature: F-094 Retrieve ChallengeQuestions for Notice of Change
+
+  Background:
+    Given an appropriate test context as detailed in the test data source,
+    And a user [MCA System - with an IDAM role of caseworker CAA],
+
+  @S-094.1
+  Scenario: Must successfully return the contents of the ChallengeQuestion definition
```
Comment (MSancaktutar) [inline aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/F-094.feature:9]: Suggestion: `must successfully return challenge questions of a correctly configured case type`

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
just some suggestions. feel free to ignore.

**Pull request merged**